doc/user/content/transform-data/patterns/percentiles.md

```diff
@@ -10,134 +10,306 @@ menu:
-
-```mzsql
-SELECT bucket AS approximate_percentile
-FROM hdr_distribution
-WHERE cumulative_distribution >= 0.9
-ORDER BY cumulative_distribution
-LIMIT 1;
-```
+   {{</ tab >}}
 
-As with histograms, increase query performance by creating an index on the `cumulative_distribution` column.
+   {{< tab "psql" >}}
 
 ```mzsql
-CREATE INDEX hdr_distribution_idx ON hdr_distribution (cumulative_distribution);
-```
-
-
-## Examples
+-- precision for the representation of the mantissa in bits
+\set precision 4
 
-```mzsql
-CREATE TABLE input (value BIGINT);
+CREATE VIEW hdr_histogram AS
+WITH
+  input_parts AS (
+    SELECT
+      CASE WHEN value = 0 THEN NULL
+          ELSE trunc(log(2, abs(value)))::int
+      END AS exponent,
+      CASE WHEN value = 0 THEN NULL
+          ELSE value / pow(2.0, trunc(log(2, abs(value)))::int)
+      END AS mantissa
+    FROM input
+  ),
+  buckets AS (
+    SELECT
+      trunc(mantissa * pow(2.0, :precision)) / pow(2.0, :precision)
+        * pow(2.0, exponent)
+        AS bucket
+    FROM input_parts
+  )
+SELECT
+  COALESCE(bucket, 0) AS bucket,
+  count(*) AS frequency
+FROM buckets
+GROUP BY bucket;
 ```
-
-Let's add the values 1 to 10 into the `input` table.
+   {{</ tab >}}
+   {{</ tabs >}}
+
+1. Create a view `hdr_distribution` to calculate the cumulative frequency/count
+   and the cumulative distribution. The view definition uses a cross join on the
+   `hdr_histogram` view. The distribution is calculated by dividing the
+   cumulative frequency (the count of all values up to and including that
+   bucket) by the total counts for all bucket values.
+
+   ```mzsql
+   CREATE VIEW hdr_distribution AS
+   SELECT
+     h.bucket,
+     h.frequency,
+     sum(g.frequency) AS cumulative_frequency,
+     sum(g.frequency) / (SELECT sum(frequency) FROM hdr_histogram) AS  cumulative_distribution
+   FROM hdr_histogram g, hdr_histogram h
+   WHERE g.bucket <= h.bucket
+   GROUP BY h.bucket, h.frequency;
+   ```
+
+1. You can then query `hdr_distribution` by the `cumulative_distribution` field
+   to return _approximate_ percentiles. More precisely, the query returns the
+   lower bound for the percentile (the next larger bucket represents the upper
+   bound).
+
+   For example, the following query returns the lower bound for the 90-th
+   percentile.
+
+   ```mzsql
+   SELECT bucket AS approximate_percentile
+   FROM hdr_distribution
+   WHERE cumulative_distribution >= 0.9
+   ORDER BY cumulative_distribution
+   LIMIT 1;
+   ```
+
+1. To keep the `hdr_distribution` view up to date as new data comes into
+   `input`, you can create an index on the view. To optimize for percentile
+   queries, index on the `cumulative_distribution` column to support [point
+   lookup on the index](/concepts/indexes/#point-lookups):
+
+   ```mzsql
+   CREATE INDEX hdr_distribution_idx ON hdr_distribution (cumulative_distribution);
```

Comment:
Same comment as above: I suspect this index will not help other than by maintaining the result (the `cumulative_distribution` column is probably hard to guess).

Reply:
Removed in patch 75f32afbf3839b0dc7b5f84321847c6c88161bff